### DISCUSSION TITLE
Configuration file PostUp Parameter not supported

### DISCUSSION THREAD

**Discussion by luweijun1992** [wiresock/WireSockUI#45] [Ideas]
PostUp Parameter not supported
![image](https://github.com/wiresock/WireSockUI/assets/58923704/f6e5406a-f6fd-4709-8fd8-90879f41c432)

**Comment by wiresock** [comment]
Yes, the WireSock VPN Client does not support PostUp/PostDown scripts. Could you please specify your requirements or the particular use case for this feature?

**Comment by luweijun1992** [comment]
I'm connected to multiple vpns. The interface metric is different, causing the system to use another VPN's DNS server.
Caused me to use the script
`
PostUp = PowerShell -Command "Set-NetIPInterface -InterfaceAlias '%WIREGUARD_TUNNEL_NAME%' -InterfaceMetric 0
`

**Comment by XFox111** [comment]
I can add another use case (though it's in favor of `PreUp`):

Recently I had an issue, where there was a DPI filter that was blocking all WG traffic by protocol (not just traffic to a specific server). So, in order to bypass this filter, I had to send some garbage data first before connecting and `PreUp` option would come out handy for me.

Plus, it is supported by official WG client, so if WireSock basically is a better Wireguard, why it should not to as well?

**Comment by wiresock** [comment]
Well, you might have a point, and it seems worthwhile to implement. I'm currently on vacation and could have some free time during the week. At first glance, it doesn't appear too complex.

> Recently I had an issue, where there was a DPI filter that was blocking all WG traffic by protocol (not just traffic to a specific server). So, in order to bypass this filter, I had to send some garbage data first before connecting and PreUp option would come out handy for me.

WireSock includes a SOCKS5 feature specifically designed for the handshake process to bypass Wireguard blocks. To the best of my knowledge, this functionality is still operational in Russia. Additionally, it is highly likely to work in Iran too, as evidenced by a significant number of web-site visitors from that country. I am intending to further develop and enhance this feature.

**Reply by XFox111** [reply]
> WireSock includes a SOCKS5 feature specifically designed for the handshake process to bypass Wireguard blocks.

Yeah, but it requires additional server configuration which doesn't worth it for me (yet)

**Reply by luweijun1992** [reply]
Is this feature supported ?

**Comment by wiresock** [comment]
This feature is supported in WireSock VPN Client v1.4.5 and WireSockUI v0.2.6.
